keps/sig-api-machinery/20190425-structural-openapi.md

```diff
@@ -314,7 +315,7 @@ We add the following additional validations for CRDs based on the representation
  
 1. **structurality:** both `ForbiddenGenerics` and `ForbiddenExtensions` only have zero values, with the two exceptions of (4).
 2. **completeness:** the schema is _structurally complete_
-3. **RawExtension:** for every schema with `x-kubernetes-embedded-resource: true`, `x-kubernetes-preserve-unknown-fields: true` and `type: object` are set
+3. **RawExtension:** for every schema with `x-kubernetes-embedded-resource: true`, `type: object` is set, and either `properties` is non-empty or `x-kubernetes-preserve-unknown-fields: true` is set.
```

Comment:
s/is non-empty/is specified/

Reply:
Can we guarantee that "is specified" in the empty case (= empty CR, doesn't make much sense) survives protobuf?

Reply:
```yaml
properties:
  embedded:
    x-kubernetes-embedded-resource:
    properties: {}
```

This might end up as

```yaml
properties:
  embedded:
    x-kubernetes-embedded-resource:
```

after many roundtrips.

Reply:
"... and either properties are specified..." would be fine.

Reply:
updated